From 646a154d492f6c9eeb6f6a4b4b5baeea1d074da2 Mon Sep 17 00:00:00 2001
From: Denys Stolbov <d.stolbov@inango-systems.com>
Date: Wed, 21 May 2025 17:49:10 +0300
Subject: [PATCH] link_metrics_collection: Explicitly send the beacon request

Explicitly send another beacon request with the values packed in
beacon_metrics_query. With the introduction of beacon reports multi-channel support we send only requests
for the channels that are packed inside chan_report_list, but not for
the channel inside the beacon metrics query. The 4.7.9 test expects the
beacon request with the beacon metrics query.

PPM-3318

Signed-off-by: Denys Stolbov <d.stolbov@inango-systems.com>
---
 .../beerocks/slave/tasks/link_metrics_collection_task.cpp   | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/agent/src/beerocks/slave/tasks/link_metrics_collection_task.cpp b/agent/src/beerocks/slave/tasks/link_metrics_collection_task.cpp
index 57561c9e21..962cc633c7 100644
--- a/agent/src/beerocks/slave/tasks/link_metrics_collection_task.cpp
+++ b/agent/src/beerocks/slave/tasks/link_metrics_collection_task.cpp
@@ -431,6 +431,12 @@ bool LinkMetricsCollectionTask::schedule_beacon_metrics_query(
         }
     }
 
+    // USED IN TESTS: Explicitly send the query with the values from dev_send_1905
+    sBeaconMetricsQuery::sChanReport new_report = {};
+    new_report.op_class                         = params.op_class;
+    new_report.channel                          = params.channel;
+    beacon_params.chan_report_list.push_back(new_report);
+
     auto request_out = message_com::create_vs_message<
         beerocks_message::cACTION_APMANAGER_MULTI_CHAN_BEACON_11K_REQUEST>(m_cmdu_tx, mid);
     if (request_out == nullptr) {
-- 
GitLab

